taichi/backends/metal/shaders/runtime_structs.metal.h

```diff
@@ -102,6 +128,21 @@ STR(
 
       Extractor extractors[kTaichiMaxNumIndices];
     };
+
+    struct ElementCoords { int32_t at[kTaichiMaxNumIndices]; };
+
+    struct ListgenElement {
+      ElementCoords coords;
+      // Memory offset from a given address.
+      // * If in_root_buffer() is true, this is from the root buffer address.
+      // * O.W. this is from the |id|-th NodeManager's |elem_idx|-th element.
```

Comment:
This is not implemented in this PR yet (therefore the placeholder `in_root_buffer()` always returns `true` at this point).